### PR TITLE
Notification panel for in progress/completed async task in admin panel

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
New notification system for admin panel to check task in progress and completed tasks
Now we will have a bell icon in admin panel header

Screenshot 2020-09-28 at 1 36 32 AM

It will have badge and counter for unread notifications:

Screenshot 2020-09-28 at 1 37 52 AM

Notifications will marked as unread when user will open the notification panel.
New Notifications will have blue dot in the end.

Screenshot 2020-09-28 at 1 39 21 AM

Notification panel consist of 2 parts:

* Task in progress
* completed tasks

user will be able to clear/remove completed task notifications but task in progress can't be removed.
They will be automatically shifted from in progress to completed when task is finished.

Future Implementations/Improvements:

1) Right now I am soft deleting the notifications user do not wish to see so in future we can have view older notifications to open a link or a panel with older notifications.

2) Data is json field of notifications. I am purposely adding profile id linked to task in data. In future we can also have a panel in which we show user's profile updates done by other user's of same profile
For example user will be able to see/receive notification for uploaded data to a profile by some other admin

3) Using ASGI server to implement notification updates using sockets or django channels.
right now we are sending request to server in every 10 seconds to check for new notifications.

## How to test it locally

I am having a weird issue on my local.
I am setting `NOTIFICATIONS_NOTIFICATION_MODEL = 'general.Notification'` in common.py config file
Still getting error for duplicate classes.
If you get this error while running on local, Please copy this setting and paste it in your local.py config
It should get it working.

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by goyal1092** [comment]
[If multiple Admin windows are open, a notification message is only shown in one of them which means that you admin may miss it if they are not looking at that particular window/tab](https://trello.com/c/eWu4je1f/444-if-multiple-admin-windows-are-open-a-notification-message-is-only-shown-in-one-of-them-which-means-that-you-admin-may-miss-it-if)

**Comment by goyal1092** [comment]
[It would be helpful for the data admin to have a consolidated view of which backend processes are currently taking place](https://trello.com/c/K7SmPx7V/449-it-would-be-helpful-for-the-data-admin-to-have-a-consolidated-view-of-which-backend-processes-are-currently-taking-place)

File: wazimap_ng/datasets/hooks.py [hunk 1]
```diff
@@ -3,152 +3,51 @@
...
-    task_list.append(task)
-    session["task_list"] = task_list
+        notification = Notification.objects.filter(task_id=task.id).first()
+
+        if notification:
+            notification.level = notification_type
+            verb = get_notification_verb(task_type, notification_type)
+            if verb:
+                notification.verb = verb
+            notification.unread = True
```
Comment (milafrerichs) [inline wazimap_ng/datasets/hooks.py:52]: if I read this correctly instead of generating a new notification you update the existing one to be read? 

Why not add a new one? What is the rationale?

**Review by milafrerichs — CHANGES_REQUESTED** [review]
you reimplemented the complete `django-notifications` package. 

that will just add a lot of complexity to maintain this. please just add the `taks_id` as a custom field as json data. 
It is currently only used to mark the notification as read once the task is complete. 
A second notification should be fine although you could mark it as read with the id in the json data as well.

File: wazimap_ng/datasets/hooks.py (hunk 1 above)
Comment (goyal1092) [reply wazimap_ng/datasets/hooks.py:52]: Requirement for the task was to show task in progress.

So to complete that what we do is:
* Start a notification and mark it in progress.
* In progress tasks are showed in separate area in notifications.
* When Django-Q is done processing that task fetch notification linked to that specific task and change language of the task that is verb and set it as unread.
* Change level of task to the required on: error, warning or success and it will be removed from in progress and shifted to other notification.
* Where a dot will appear so user will know this is new notification for task completed
